OOP/oop_home_task.py

```diff
@@ -0,0 +1,475 @@
+        """
+        expiration_date = input("Please enter expiration date (yyyy-mm-dd): \n")
+        try:
+            datetime.datetime.strptime(expiration_date, '%Y-%m-%d')
+            return expiration_date
+        except ValueError:
+            print('Incorrect date. Please use this format only: yyyy-mm-dd.')
+            exit(-1)
+
+    @staticmethod
+    def ask_author():
+        """
+        Method for entering text author.
+        @return: author of publication entered by user
+        """
+        author = input("Please enter your name: \n")
+        return author
+
+
+class News(Publication):
+    """
+    Class that represents the news publication type. Inherits Publication class.
+    """
+    def __init__(self, publication_date, city, publication_text, publication_type=1):
+        """
+        Method is constructor for news class
+        @param publication_date: date of publication
+        @param publication_type: type of publication
+        @param city: city where something happened
+        @param publication_text: text of publication
+        """
+        self.news_splitter = 'News -------------------------'
+        super().__init__(city, publication_type, publication_date, publication_text)
+
+    def publish_news(self):
+        """
+        Blank method for publishing of news that will be overwritten
+        @return: nothing
+        """
+        pass
+
+
+class Advertisement(Publication):
+    """
+    Class that represents the Advertisement publication type. Inherits Publication class.
+    """
+    def __init__(self, publication_date, expiration_date, publication_text, publication_type=2, city=''):
+        """
+        Method is constructor for Advertisement class
+        @param publication_date: date of publication
+        @param expiration_date: date of Advertisement expiration
+        @param publication_text: text of publication
+        @param publication_type: type of publication
+        @param city: city where something happened
+        """
+        self.advertisement_splitter = 'Private Ad -------------------'
+        self.expiration_date = expiration_date
+        super().__init__(publication_date, publication_type, city, publication_text)
+
+    @staticmethod
+    def calculate_expiration_days(expiration_date):
+        """
+        Method for expiration days calculation. Catches TypeError exception.
+        @param expiration_date: date of Advertisement expiration
+        @return: count of days from today to expiration date
+        """
+        try:
+            expiration_days = (datetime.date.today() - datetime.date.fromisoformat(expiration_date))*-1
+            return expiration_days.days
+        except TypeError:
+            print('Expiration days calculation error.')
+
+    def publish_advertisement(self):
+        """
+        Blank method for publishing of Advertisement that will be overwritten
+        @return: nothing
+        """
+        pass
```

Comment:
But if we delete it nothing will change, right? What is a sense in this empty method then?
Here I see 2 options:
1) Delete it.
2) Go deeper in OOP and learn about abstract classes and methods. There is implementation of it in Python in ABC module. If you make this class as abstract it will make sense.

Reply:
Deleted.